documentation/design/iot/README.adoc

```diff
@@ -1,38 +1,148 @@
++
+This is an implicit assignment. An IoT project resides in a Kubernetes
+namespace, which is assigned to a messaging tenant, which is assigned
+to a messaging infrastructure.
+
+*Target persona:* IoT service admin
+
+*Proposal:* Change the name from `IoTConfig` to `IoTInfrastructure`
+when we make allow for multiple IoT infrastructures.
+
+=== Hono services
+
+Each IoT infrastructure has a set of Hono services (protocol adapters,
+device registry, …) which will be shared between all IoT projects,
+assigned to this infrastructure.
+
+=== Protocol adapter addressing
+
+The protocol adapters will directly connect to the messaging
+infrastructure, using an (to be defined) internal
+authentication/authorization mechanism.
+
+For this to work, the protocol adapters must be able to prefix
+the address resources with the _vhost_ prefix. The _vhost_ prefix
+is the Kubernetes namespace.
+
+*Example:* Assuming an IoT project `iot1` in the namespace `ns1`.
+The operator would request an address `telemetry/ns1.iot1` in the
+namespace `ns1`. As the protocol adapters would connect to the
+messaging infrastructure, they would prefix the address with the
+namespace and would use the address `ns1/telemetry/ns1.iot1` instead.
+
+IMPORTANT: Currently we are using the QPID dispatch router sidecar to
+split the single AMQP connection coming from Hono protocol adapters
+into multiple connections for the different address spaces. This
+sidecar setup would be removed.
+
+IMPORTANT: This requires an upstream change in Hono, so that we
+have the ability to prefix the addresses in Hono with a custom
+(tenant specific) prefix.
+
 == IoT Project CRD
 
-An "IoT project" is a logical entity which scopes a set of devices into a single, IoT specific, namespace.
+[.lead]
+An "IoT project" is a logical entity, which scopes a set of devices into a single, IoT specific, namespace.
 
 An IoT project:
 
 * Is part of a Kubernetes namespace
 * Has a name which adheres to the requirements of `.metadata.name`
 * Translates into a Hono tenant name by: `<namespace>.<name>`
 * Can be described in a Kubernetes CRD
-* Connects to one EnMasse standard address space
-* Has its own set of addresses ("telemtry/", "event/", ...) inside that address space
+* Connects implicitly to one EnMasse IoT infrastructure (IoTConfig)
+* Manages its own set of addresses ("telemetry/", "event/", ...)
+* Requires a _messaging tenant_ to be created first
+
+IMPORTANT: This will remove the three current _downstream strategies_
+and remain with only the _managed_ one. The "external" strategy can
+be achieved by using _EnMasse connectors_.
+
+*Target persona:* IoT tenant
+
+=== Missing messaging tenant
+
+*Aspect:* Getting started
+
+If an IoT project gets created in a namespace which does not have
+a _message tenant_ created, then the IoT project will:
+
+* Get the field `.status.phase` set to `Failed`
+* Provide a helpful message in the field `.status.phaseReason`
```

Comment:
Again pointing out an inconsistency in iot vs enmasse with .status.message vs .status.phaseReason, where we should agree on one of them.

Reply:
Taking at look at the Kubernetes docs:

> Some resources in the v1 API contain fields called `phase`, and associated `message`, `reason`, and other status fields.

So I think it should be either `reason` or `message`, but not `phaseReason`. If we already have `message` elsewhere, I would propose to switch to `message` here as well.

Reply:
Done.